src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -60,15 +55,12 @@ public void emptyBasket() {
 
     public List<Product> removeByName(String name) {
         List<Product> result = new LinkedList<>();
-        Iterator<Product> iterator = products.iterator();
-        Product element;
-        while (iterator.hasNext()) {
-            element = iterator.next();
-            if (element.getName().equals(name)) {
-                result.add(element);
-                iterator.remove();
+        for (Map.Entry<String, ArrayList<Product>> productList : products.entrySet()) {
```

Comment:
```suggestion
        for (Map.Entry<String, List<Product>> productList : products.entrySet()) {
```
не забываем про полиморфизм